src/shared/filesystemUtilities.ts

```diff
@@ -162,3 +162,26 @@ export async function hasFileWithSuffix(dir: string, suffix: string, exclude?: v
     const matchedFiles = await vscode.workspace.findFiles(searchFolder, exclude, 1)
     return matchedFiles.length > 0
 }
+
+/**
+ * Searches directory and all sub-directories for a filename.
+ * @param dir Directory to search
+ * @param fileName Name of file to locate
+ * @returns  The absolute path if found.
+ */
+export async function findFile(dir: string, fileName: string): Promise<string | undefined> {
+    const files = await readdir(dir)
+    const subDirs = []
+    for (const file of files) {
+        const filePath = path.join(dir, file)
+        if (filePath === path.join(dir, fileName)) {
+            return filePath
+        }
+        if ((await stat(filePath)).isDirectory()) {
+            subDirs.push(filePath)
+        }
+    }
+    for (const dir of subDirs) {
+        findFile(dir, fileName)
```

Comment:
This doesn't do anything, so `.application.json` weren't found on Cloud9 if deeper than one level of nesting.